wgpu-core/src/resource.rs

```diff
@@ -778,17 +779,20 @@ impl Buffer {
             })
         };
 
-        let mut pending_writes = device.pending_writes.lock();
-        if pending_writes.contains_buffer(self) {
-            pending_writes.consume_temp(temp);
-        } else {
-            let mut life_lock = device.lock_life();
-            let last_submit_index = life_lock.get_buffer_latest_submission_index(self);
-            if let Some(last_submit_index) = last_submit_index {
-                life_lock.schedule_resource_destruction(temp, last_submit_index);
+        if let Some(queue) = device.get_queue() {
+            let mut pending_writes = queue.pending_writes.lock();
+            if pending_writes.contains_buffer(self) {
+                pending_writes.consume_temp(temp);
+                return Ok(());
             }
```

Comment:
thought: This is an interesting change, because previously, `device.pending_writes` had a lock region that overlapped the `life_lock` region. Now, they're disparate.

Reply:
question: Actually, I'm surprised this doesn't result in a deletion of an entry in the `QUEUE_PENDING_WRITES` entry in `wgpu-core/src/lock/rank.rs`. I'm guessing that this wasn't done because there are other cases where the `life_lock` is acquired within the region of a `pending_writes` lock?

Reply:
Maybe, but I didn't check.

Reply:
One of the next commits that moves the `LifetimeTracker` into the `Queue` changes this back to the previous shape.